### ISSUE TITLE
pandas.core.base.DataError: No numeric types to aggregate

### ISSUE THREAD

**Issue by ishgupta** [pandas-dev/pandas#34403]
aggregating a boolean fields doesn't allow averaging the data column in the latest version. Is there a new alternative of doing this for boolean attributes, or it should only be handled by transforming it to a int/float only?

data[ group_fields + [ bool_field ]].groupby( group_fields ).mean() produces the error mentioned in subject.

**Comment by MarcoGorelli** [comment]
Thanks for the report - to expedite resolution, could you include a reproducible example https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports ?

**Comment by dsaxton** [comment]
I think this should be fixed after https://github.com/pandas-dev/pandas/pull/34056

**Comment by TomAugspurger** [comment]
Sounds like it. Let us know if it's not fixed by #34056 @ishgupta.

**Comment by ishgupta** [comment]
Hi @TomAugspurger, thanks for sharing that..
I refer to the similar issue mentioned in [#34056](https://github.com/pandas-dev/pandas/pull/34056), but with another extension..

with the below code, sum works fine on a boolean column, but mean() leads to error.
[ @MarcoGorelli ]
df = pd.DataFrame({"a": [1] * 50_000 + [2] * 50_000, "b": pd.array([True] * 100_000)})
df.groupby("a").sum()    # works fine
df.groupby("a").mean()    # DataError: No numeric types to aggregate

**Comment by dsaxton** [comment]
@ishgupta I just checked to confirm and this does indeed work on master:

```python
[ins] In [3]: df = pd.DataFrame({"a": [1] * 50_000 + [2] * 50_000, "b": pd.array([True] * 100_000)})                                                                                                         

[ins] In [4]: df.groupby("a").sum()                                                                                                                                                                          
Out[4]: 
       b
a       
1  50000
2  50000

[ins] In [5]: df.groupby("a").mean()                                                                                                                                                                         
Out[5]: 
      b
a      
1  True
2  True

[ins] In [6]: pd.__version__                                                                                                                                                                                 
Out[6]: '1.1.0.dev0+1758.g035e1fe83'
```

**Comment by ishgupta** [comment]
@dsaxton Thank you for taking time out to respond my query.
Actually, I am using the latest version of pandas available, which is 1.0.4, and yes, it is older than the development version you're using. :)

Is it possible for you to guide me, if a stable release of v1.1.0 available on PyPI ?

**Comment by TomAugspurger** [comment]
Not on PyPI. Nightly wheels are available at
https://anaconda.org/scipy-wheels-nightly/pandas.

It'll be a few weeks before 1.1
